test_entropy.py

```diff
@@ -20,71 +20,79 @@
-        self.assertEqual(e.entropy_to_bip39(entropy, 12),
-                         'girl mad pet galaxy egg matter matrix prison refuse sense ordinary nose')
-
-        entropy = e.bip32_xprv_to_entropy("m/83696968'/39'/0'/18'/0'", xprv)
-        self.assertEqual(entropy[:24].hex(), '938033ed8b12698449d4bbca3c853c66b293ea1b1ce9d9dc')
-        self.assertEqual(e.entropy_to_bip39(entropy, 18),
-                         'near account window bike charge season chef number sketch tomorrow excuse sniff circle vital hockey outdoor supply token')
-
-        entropy = e.bip32_xprv_to_entropy("m/83696968'/39'/0'/24'/0'", xprv)
-        self.assertEqual(entropy[:32].hex(), 'ae131e2312cdc61331542efe0d1077bac5ea803adf24b313a4f0e48e9c51f37f')
-        self.assertEqual(e.entropy_to_bip39(entropy, 24),
-                         'puppy ocean match cereal symbol another shed magic wrap hammer bulb intact gadget divorce twin tonight reason outdoor destroy simple truth cigar social volcano')
-
-
-def __main__():
-    unittest.main()
-
+import pytest
+
+XPRV = 'xprv9s21ZrQH143K2LBWUUQRFXhucrQqBpKdRRxNVq2zBqsx8HVqFk2uYo8kmbaLLHRdqtQpUm98uKfu3vca1LqdGhUtyoFnCNkfmXRyPXLjbKb'
+
+def test_mnemonic():
+    e = bipentropy.BIPEntropy()
+    mnemonic = 'install scatter logic circle pencil average fall shoe quantum disease suspect usage'
+    test = e.bip39_mnemonic_to_entropy("m/83696968'/0'/0'", mnemonic)
+    expected = '71356cdf2f6851c0499b47fe16ef121883f8c58e7d6fb4c33a9017df71f3b0fe21f92d043ee4f19676384a3d943554904caca131269dcb84151ecb2f7ca31902'
+    assert test.hex() == expected
+
+    # with password
+    test = e.bip39_mnemonic_to_entropy("m/83696968'/0'/0'", mnemonic, 'TREZOR')
+    expected = '5c56a2a19ebe8f86c4cbb788dd264bd96387dac2047dac799c51fb6218da0513da44bc0f4603815cc8c8c456dbd3aae79e334fb19dbeffc43fc236d58368ebdb'
+    assert test.hex() == expected
+
+def test_xprv_to_entropy():
+    e = bipentropy.BIPEntropy()
+    test = e.bip32_xprv_to_entropy("m/83696968'/0'/0'", XPRV)
+    expected = '71356cdf2f6851c0499b47fe16ef121883f8c58e7d6fb4c33a9017df71f3b0fe21f92d043ee4f19676384a3d943554904caca131269dcb84151ecb2f7ca31902'
+    assert test.hex() == expected
+
+
+def test_entropy_to_mnemonic():
+    e = bipentropy.BIPEntropy()
+    entropy = e.bip32_xprv_to_entropy("m/83696968'/0'/0'", XPRV)
+
+    words12 = 'illness problem daughter gain lunar then chapter happy wrap resist setup corn'
+    assert e.entropy_to_bip39(entropy, 12) == words12
+
+    words15 = 'illness problem daughter gain lunar then chapter happy wrap resist setup country display glare debate'
+    assert e.entropy_to_bip39(entropy, 15) == words15
+
+    words24 = 'illness problem daughter gain lunar then chapter happy wrap resist setup country display glare delay pupil regular border piano cook warfare what sentence supreme'
+    assert e.entropy_to_bip39(entropy, 24) == words24
+
+def test_wif_from_entropy():
+    # PDG: not sure about this?
```

Comment:
Can you elaborate?

Reply:
I haven't tested this specific case against Coldcard, that's all.